File: app/src/main/java/siergo_o/onlinernews/MainActivity.java
package siergo_o.onlinernews;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabLayout;

import java.util.Objects;

import siergo_o.onlinernews.view.FragmentNewsHolder;


public class MainActivity extends AppCompatActivity {



    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);



        ViewPager viewPager = findViewById(R.id.viewpager);
        viewPager.setOffscreenPageLimit(3);
        PagerAdapter pagerAdapter = new PagerAdapter(getSupportFragmentManager(), MainActivity.this);
        viewPager.setAdapter(pagerAdapter);

        // Give the TabLayout the ViewPager
        TabLayout tabLayout = findViewById(R.id.tablayout);
        tabLayout.setupWithViewPager(viewPager);

        // Iterate over all tabs and set the custom view
        for (int i = 0; i < tabLayout.getTabCount(); i++) {
            TabLayout.Tab tab = tabLayout.getTabAt(i);
            Objects.requireNonNull(tab).setCustomView(pagerAdapter.getTabView(i));
        }

    }

    @Override
    public void onResume() {
        super.onResume();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        return super.onOptionsItemSelected(item);
    }



    class PagerAdapter extends FragmentPagerAdapter {

        String[] tabTitles = {"Техно", "Люди", "Авто"};
        TextView tv;
        View tab;
        Context context;

        public PagerAdapter(FragmentManager fm, Context context) {
            super(fm);
            this.context = context;
        }

        @Override
        public int getCount() {
            return tabTitles.length;
        }

        @Override
        public Fragment getItem(int position) {

            switch (position) {
                case 0:
                    return new FragmentNewsHolder("https://tech.onliner.by/", position);
                case 1:
                    return new FragmentNewsHolder("https://people.onliner.by/", position);
                case 2:
                    return new FragmentNewsHolder("https://auto.onliner.by/", position);
            }

            return null;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            // Generate title based on item position
            return tabTitles[position];
        }

        @SuppressLint("InflateParams")
        View getTabView(int position) {

            tab = LayoutInflater.from(context).inflate(R.layout.custom_tab, null);
            tv = tab.findViewById(R.id.title_tab_text);
            tv.setText(tabTitles[position]);
            return tab;
        }
    }

}

File: app/src/main/java/siergo_o/onlinernews/MainActivity.kt
package siergo_o.onlinernews;

import android.annotation.SuppressLint
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.iterator
import androidx.core.view.size
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import siergo_o.onlinernews.view.FragmentNewsHolder
import java.util.*


class MainActivity: AppCompatActivity() {


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_main);

        val viewPager: ViewPager = findViewById(R.id.viewpager);
        viewPager.offscreenPageLimit = 3
        val pagerAdapter = PagerAdapter(supportFragmentManager, this);
        viewPager.adapter = pagerAdapter

        val tabLayout: TabLayout = findViewById(R.id.tablayout);
        tabLayout.setupWithViewPager(viewPager);

        for (i in 0..tabLayout.size) {
            val tab = tabLayout.getTabAt(i)
            tab?.customView = pagerAdapter.getTabView(i)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    inner class PagerAdapter(fm: FragmentManager, private val context: Context): FragmentPagerAdapter(fm) {

        private val tabTitles = listOf("Техно", "Люди", "Авто")
        private var tv: TextView? = null
        private var tab: View? = null

        override fun getCount(): Int = tabTitles.size

        override fun getItem(position: Int): Fragment =
                when (position) {
                    0 -> FragmentNewsHolder("https://tech.onliner.by/")
                    1 -> FragmentNewsHolder("https://people.onliner.by/")
                    2 -> FragmentNewsHolder("https://auto.onliner.by//")
                    else -> FragmentNewsHolder("https://tech.onliner.by/")
                }

        override fun getPageTitle(position: Int): CharSequence? = tabTitles[position]

        @SuppressLint("InflateParams")
        fun getTabView(position: Int): View? {
            tab = LayoutInflater.from(context).inflate(R.layout.custom_tab,null)
            tv = tab?.findViewById(R.id.title_tab_text)
            tv?.text = tabTitles[position];
            return tab
        }
    }

}


